docs/architecture/adr-108-e2e-abci++.md

```diff
@@ -0,0 +1,296 @@
+The idea was to do this at the Application side. Every time the Application 
+receives a request, it logs it.
+
+**Implementation**
+
+The rationale behind this part of the implementation was to log the request concisely and use the existing structures as much as possible. 
+
+Whenever an ABCI++ request is made, the application will create `abci.Request` (`abci` stands for `"github.com/cometbft/cometbft/abci/types"`) and log it.  The example is below.  
+
+```go
+func (app *Application) InitChain(_ context.Context, req *abci.RequestInitChain) (*abci.ResponseInitChain, error) {
+
+	r := &abci.Request{Value: &abci.Request_InitChain{InitChain: &abci.RequestInitChain{}}}
+	err := app.logAbciRequest(r)
+	if err != nil {
+		return nil, err
+	}
+    
+	...
+}
+```
+Notice here that we create an empty `abci.RequestInitChain` object while we can also use the one passed to the `InitChain` function. The reason behind this is that, at the moment, we do not need specific fields of the request; we just need to be able to extract the information about the request type. For this, an empty object of a particular type is enough. 
+
+The `app.logABCIRequest(r)` function is a new function implemented in the same file (`test/e2e/app/app.go`). If the `ABCIRequestsLoggingEnabled` flag is set to `true`, set automatically when ABCI++ tests are enabled, it logs received requests. The full implementation is the following: 
+
+```go
+func (app *Application) logABCIRequest(req *abci.Request) error {
+	if !app.cfg.ABCIRequestsLoggingEnabled {
+		return nil
+	}
+	s, err := GetABCIRequestString(req)
+	if err != nil {
+		return err
+	}
+	app.logger.Info(s)
+	return nil
+}
+```
+
+`GetABCIRequestString(req)` is a new method that receives a request and returns its string representation. The implementation and tests for this function and the opposite function `GetABCIRequestFromString(req)`
+that returns `abci.Request` from the string are provided in files `test/e2e/app/log.go` and `test/e2e/app/log_test.go`, respectively. To create a string representation of a request, we first marshal the request via `proto.Marshal()` method and then convert received bytes in the string using `base64.StdEncoding.EncodeToString()` method. In addition, we surround the new string with `abci-req` constants so that we can find lines with ABCI++ request more easily. The code of the method is below: 
+
+```go
+func GetABCIRequestString(req *abci.Request) (string, error) {
+	b, err := proto.Marshal(req)
+	if err != nil {
+		return "", err
+	}
+	reqStr := base64.StdEncoding.EncodeToString(b)
+	s := ABCI_REQ + reqStr + ABCI_REQ
+	return s, nil
+}
+```
+
+*Note:* At the moment, we are not compressing the marshalled request before converting it to `base64` `string` because we are logging the empty requests that take at most 24B. However, if we decide to log the actual requests in the future, we might want to compress them. Based on a few tests, we observed that the size of a request can go up to 7KB.  
```

Comment:
```suggestion
*Note:* At the moment, we are not compressing the marshalled request before converting it to `base64` `string` because we are logging the empty requests that take at most 24 bytes. However, if we decide to log the actual requests in the future, we might want to compress them. Based on a few tests, we observed that the size of a request can go up to 7KB.  
```